src/Components/Dashboard/dashboard.js

```diff
@@ -0,0 +1,90 @@
+import React, { useCallback, useEffect, useState } from "react";
+import moment from 'moment';
+import '../../stylesheets/dashboard.css'
+import {Col, Spin, Card, Tooltip } from 'antd';
+
+export const Dashboard = () => {
+    const [pictureDayInfo, setPictureDayInfo] = useState([])
+    const [fetchingApi, setFetchingApi] = useState(true);
+    const [erroMsg, setErrorMsg] = useState(false);
+    const [unexpectedError, setUnexpectedError] = useState(false)
+    const [date, setDate] = useState(moment(new Date()).format("YYYY-MM-DD"))
+
+
+    const fetchMyAPI = useCallback(async () => {
+        if(moment(date).isAfter(moment(new Date()).format("YYYY-MM-DD"))===true || moment('1995-06-16').isSameOrAfter(date)===true ){
+            setErrorMsg(true)
+        }else{
+            const apiKey = 'wOjD8BmGH4ZViJyv2OpeZlKfSaA24XkEcvmyipkF';
+            let response = await fetch(`https://api.nasa.gov/planetary/apod?api_key=${apiKey}&start_date=${date}&end_date=${date}`)
```

Comment:
Your app is looking good with the current changes but it's a good practice to abstract the fetching logic and states related to a custom hook so with this approach the components are more stateless and easier to apply tests.
Feel free to not implement the changes now, but it's a best practice to have custom hooks.
[Article How to fetch data with React Hooks](https://www.robinwieruch.de/react-hooks-fetch-data/)
[CodeSandbox useFetch](https://codesandbox.io/s/usereducer-example-71pbj8?file=/src/App.js)